sqld/src/http/mod.rs

```diff
@@ -169,75 +155,60 @@ async fn handle_upgrade(
         .await;
 
     match response_rx.await {
-        Ok(response) => response,
-        Err(_) => Response::builder()
-            .status(hyper::StatusCode::SERVICE_UNAVAILABLE)
-            .body("sqld was not able to process the HTTP upgrade".into())
-            .unwrap(),
+        Ok(response) => response.into_response(),
+        Err(_) => (
```

Comment:
This should probably be moved into its own error type but I deferred to keep the code similar as before for now.